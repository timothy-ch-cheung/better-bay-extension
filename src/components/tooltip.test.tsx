import { describe, expect, jest, test } from "@jest/globals"
import { fireEvent, render, screen, waitFor } from "@testing-library/react"
import axios from "axios"
import React from "react"

import Tooltip from "./tooltip"

jest.mock("axios")
const mockedAxios = axios as jest.Mocked<typeof axios>

jest.mock("webextension-polyfill", () => ({}))
jest.mock("@plasmohq/storage/hook", () => ({
  useStorage: jest.fn().mockReturnValue([{}])
}))

describe("Tooltip", () => {
  let tooltip

  beforeEach(() => {
    tooltip = render(<Tooltip itemId="123" />)
  })

  test("Snapshot", () => {
    expect(tooltip).toMatchSnapshot()
  })

  test("Hover Tooltip - API call successful", async () => {
    const spinner = screen.getByRole("status")
    const tooltipIcon = screen.getByText("üè∑Ô∏è")
    expect(spinner).toBeInTheDocument()

    mockedAxios.get.mockResolvedValue({
      data: {
        123: { price: "0.99", description: { colour: "Black" } }
      }
    })
    fireEvent.mouseOver(tooltipIcon)

    waitFor(() => {
      expect(screen.getByText("0.99").toBeInTheDocument())
      expect(screen.getByText("colour: Black").toBeInTheDocument())
    }).catch((error: Error) => {
      console.log(`Assertions failed [${error.message}]`)
    })
  })

  test("Hover Tooltip - API call unsuccessful", async () => {
    const spinner = screen.getByRole("status")
    const tooltipIcon = screen.getByText("üè∑Ô∏è")
    expect(spinner).toBeInTheDocument()

    mockedAxios.get.mockRejectedValue(new Error("Failed to call API"))
    fireEvent.mouseOver(tooltipIcon)

    expect(spinner).toBeInTheDocument()
  })
})
