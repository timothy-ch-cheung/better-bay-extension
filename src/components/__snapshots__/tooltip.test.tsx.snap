// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="px-0.5 py-1 text-white hover:bg-slate-300 focus:outline-none rounded-lg text-center"
        data-test="123-bb-tooltip"
        data-tooltip-placement="bottom"
        data-tooltip-target="tooltip-text-123"
        type="button"
      >
        üè∑Ô∏è
      </button>
      <div
        class="inline-block absolute invisible z-10 py-2 px-3 text-sm font-medium text-white bg-gray-900 rounded-lg shadow-sm opacity-0 transition-opacity duration-300 tooltip dark:bg-gray-700"
        id="tooltip-text-123"
        role="tooltip"
      >
        <div
          role="status"
        >
          <svg
            aria-hidden="true"
            class="mr-2 w-8 h-8 text-gray-200 animate-spin dark:text-gray-600 fill-blue-600"
            fill="none"
            viewBox="0 0 100 101"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
              fill="currentColor"
            />
            <path
              d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
              fill="currentFill"
            />
          </svg>
        </div>
        <div
          class="tooltip-arrow"
          data-popper-arrow="true"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      class="px-0.5 py-1 text-white hover:bg-slate-300 focus:outline-none rounded-lg text-center"
      data-test="123-bb-tooltip"
      data-tooltip-placement="bottom"
      data-tooltip-target="tooltip-text-123"
      type="button"
    >
      üè∑Ô∏è
    </button>
    <div
      class="inline-block absolute invisible z-10 py-2 px-3 text-sm font-medium text-white bg-gray-900 rounded-lg shadow-sm opacity-0 transition-opacity duration-300 tooltip dark:bg-gray-700"
      id="tooltip-text-123"
      role="tooltip"
    >
      <div
        role="status"
      >
        <svg
          aria-hidden="true"
          class="mr-2 w-8 h-8 text-gray-200 animate-spin dark:text-gray-600 fill-blue-600"
          fill="none"
          viewBox="0 0 100 101"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
            fill="currentColor"
          />
          <path
            d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
            fill="currentFill"
          />
        </svg>
      </div>
      <div
        class="tooltip-arrow"
        data-popper-arrow="true"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
